tests/unit/models/test_projects.py

```diff
@@ -1782,3 +1783,20 @@ def test_project_get_build_for_arch_triplet_all(self, project_yaml_data):
         arch_triplet = project.get_build_for_arch_triplet()
 
         assert not arch_triplet
+
+
+class TestProjectTransform:
+    """Test Transform the Project."""
+
+    def test_root_packages_transform(self, project_yaml_data):
```

Comment:
May be useful to have a test to ensure projects without `build-packages` and `build-snaps` are unmodified  (i.e. the `if` statement on line 181)